File: Add Code Here/C++/kadaneAlgo.cpp
#include <iostream>
using namespace std;

int k_s(int arr[], int n){
    int cs=0,ms=0,total_sum=0; // current sum and maximum sum
    for(int i=0; i<n; i++){
        total_sum += arr[i];
    }
    for(int k=0; k<n; k++){
        if (arr[k]<0){
            return total_sum;
        }
    }
    for(int i=0; i<n; i++){
        cs += arr[i];
        if(cs<0){
            cs=0;
        }
        ms = max(cs , ms);
    }
    return ms;
}

int main(){
    int arr[]={-3,-6,-9};
    int n = sizeof(arr)/sizeof(int);

    for(int i=0; i< n; i++){
        cout << arr[i] << " "; 
    }  

    cout << endl;  

   int t= k_s(arr,n);
   cout << t;

}

File: Add Code Here/Java/kadane_Algo.java
/**
 * kadane's_Algo
 * Find largest sum of the contiguous subarray of an array
 * Time complexity using Kadane algorithm- O(n)
 * contributed by code-with-Priyanshu
 */

import java.util.*;
public class kadane_Algo {

    public static void main(String[] args) {
        int[] arr={1,5,-2,8,-3,5};
        int n=6;
        int sum=0;
        int maxi=arr[0];
        for(int i=0;i<n;i++){
            sum+=arr[i];
            maxi=Math.max(maxi,sum);
            if(sum<0){
                sum=0;
            }
        }
        System.out.println("The maximum sum of contiguous subarray is "+maxi);
    }    
}